src/charm.py

```diff
@@ -42,9 +42,6 @@ def _on_config_changed(self, event):
         self.unit.status = self.app.status = self._get_status()
 
     def _get_status(self):
```

Comment:
don't we at least want a log, a warning, something if the credentials are empty?

Reply:
Not sure, because it's a perfectly valid use case; maybe a debug log could work, but I don't think it's necessary